src/Microsoft.OpenApi.OData.Reader/Common/Utils.cs

```diff
@@ -270,5 +271,29 @@ internal static void AddCustomAttributesToExtensions(this IDictionary<string, IO
 
             return false;
         }
+
+        /// <summary>
+        /// Removes the base namespace from a string value.
+        /// </summary>
+        /// <param name="value">The target string value.</param>
+        /// <param name="settings">Convert settings.</param>
+        /// <returns>The string value with the namespace removed.</returns>
+        internal static string RemoveBaseNamespace(this string value, OpenApiConvertSettings settings)
+        {
+            CheckArgumentNullOrEmpty(value, nameof(value));
+            CheckArgumentNull(settings, nameof(settings));
+
+            if (string.IsNullOrEmpty(settings.DefaultNamespace))
+            {
+                return value;
+            }
+
+            string baseNamespace = settings.DefaultNamespace.EndsWith(".")
+                ? settings.DefaultNamespace
+                : settings.DefaultNamespace + ".";
+
+            string escapedPattern = Regex.Escape(baseNamespace);
```

Comment:
parsing a regex on each call is going to get really costly. Just using a string.startswith and string.Substring if that's true is probably going to be much faster.